libsolidity/formal/CHC.cpp

```diff
@@ -1521,31 +1498,38 @@ optional<string> CHC::generateCounterexample(CHCSolverInterface::CexGraph const&
 				path.emplace_back("State: " + modelMsg);
 		}
 
-		string txCex = summaryPredicate->formatSummaryCall(summaryArgs);
 		path.emplace_back(txCex);
-
-		/// Stop when we reach the summary of the analyzed constructor.
-		if (summaryPredicate->type() == PredicateType::ConstructorSummary)
-			break;
-
-		/// Recurse on the next interface node which represents the previous transaction.
-		node = *interfaceId;
 	}
 
 	return localState + "\nTransaction trace:\n" + boost::algorithm::join(boost::adaptors::reverse(path), "\n");
 }
 
+map<unsigned, vector<unsigned>> CHC::summaryCalls(CHCSolverInterface::CexGraph const& _graph, unsigned _root)
+{
+	map<unsigned, vector<unsigned>> calls;
```

Comment:
```suggestion
	map<unsigned, vector<CexNode const*>> calls;
```
and storing ``&_graph.nodes.at(node)`` here and immediately use that at the callsite might save a handful of map lookups...